### DISCUSSION TITLE
New filter for hashes where the key is an ID of some kind with an attached value

### DISCUSSION THREAD

**Discussion by AaronLasseigne** [AaronLasseigne/active_interaction#535] [Ideas]
In #459 and #531 we see a case that isn't easily handled within ActiveInteraction. It's the situation where a has is used not as a group of information but as a lookup where the key is some unique ID and the values are... whatever.

I think I'm going to add a new filter called `lookup` to handle this. It would work by accepting any key and then validating the value if a block is given. It would be the same way the array filter works.

```ruby
lookup :cities do
  integer
end
```

Then you could pass a hash with a matching format:

```ruby
cites: { Tokyo: 37_468_000, Delhi: 28_514_000, ... } 
```

Questions:

1. Should there be a way to validate keys?

**Comment by AaronLasseigne** [comment]
Maybe `hashmap` instead of `lookup`?

**Reply by shetty-tejas** [reply]
@AaronLasseigne I do like the name `lookup` more instead of `hashmap`. As of key validation, I do think that could be helpful.

How about a syntax something like this?
```ruby
lookup[string] :cities do
  integer
end
```

So if anyone wants to use the key validation, they can by passing the data type of the key, else they can ignore.

P.S: Thinking a bit more after a while, how about something like this? The name `hashmap` would also make sense here:
```ruby
hashmap[string, integer] :cities
```
